synapse_data_warehouse/synapse/dynamic_tables/V2.29.4__userprofile_latest.sql

```diff
@@ -0,0 +1,26 @@
+-- Introduce the dynamic table
+USE SCHEMA {{database_name}}.synapse; --noqa: JJ01,PRS,TMP
+CREATE OR REPLACE DYNAMIC TABLE USERPROFILE_LATEST
+    TARGET_LAG = '1 day'
+    WAREHOUSE = compute_xsmall
+    AS 
+    WITH dedup_userprofile AS (
+        SELECT
+            *
+        FROM {{database_name}}.SYNAPSE_RAW.USERPROFILESNAPSHOT --noqa: TMP
+        WHERE
+            SNAPSHOT_DATE >= CURRENT_TIMESTAMP - INTERVAL '14 days'
```

Comment:
Should we be consistent and use the same day interval for reduplication of all snapshots?

Reply:
Consistency is a good idea. If we do this, we will have to go with `INTERVAL '30 days'` as that is the time window for `node_latest` and we can't make that one shorter.

Reply:
I second Jenny. Another reason is that the majority of the existing dynamic table is implementing the 14-days interval.

Reply:
Good idea, thank you!

Reply:
@jaymedina Another risk that we assume is if snapshotting doesn't occur for 2 weeks in a row.

That said - I don't think that happens.  We will need to determine all the tables that we need to be concerned about delete events (even if they don't all track that change type).  So do all of these below need to use interval of 30?

* Teams
* Node
* File
* ACL
* teammember?
* project settings?

Reply:
It looks like

* AR snapshots
* User group
* Userprofile

As well ( although User group and Userprofile don't capture deletions )

Another thought I just had is a lot of these tables are related to each other (file entities in `node_latest` should also be present in `file_latest`) sooo maybe for consistency as far as data availability, having all the same windows is better for the users. It would be nice to expedite having `change_type=DELETE` be available for all snapshots with a delete option on Synapse so we don't have to worry about the time window thing.

Reply:
@jaymedina @thomasyu888 @philerooski 
I did a brief check and the following snapshot tables have `change_type=DELETE`: 

- ACCESSREQUIREMENTSNAPSHOTS
- ACLSNAPSHOTS
- FILESNAPSHOTS
- NODESNAPSHOTS
- PROJECTSETTINGSNAPSHOTS

Do we agree to keep time window for dynamic tables derived from snopshots with change_type=DELETE to 30 days and other dynamic tables to 14 days?

Reply:
Yes! This workaround will do until platform team can address the missing `change_type=DELETE` for some snapshot tables. Thanks!